test/teiserver/game/balancer_server_test.exs

```diff
@@ -35,8 +40,176 @@ defmodule Teiserver.Game.BalancerServerTest do
     result = BalancerServer.calculate_team_size(team_count, players)
     assert result == 4
   end
+end
 
-  defp create_fake_players(count) do
-    1..count |> Enum.map(fn _ -> %{} end)
+defmodule Teiserver.Game.BalancerServerTest do
+  @moduledoc false
+  use Teiserver.DataCase, async: false
+  import Teiserver.Support.Polling, only: [poll_until_nil: 1]
+
+  @moduletag :balance_test
+  alias Teiserver.Game.BalancerServer
+
+  def user_fixture(), do: make_user(%{"permissions" => []})
+
+  def make_user(params \\ %{}) do
+    requested_user = %{
+      "name" => params["name"] || "Test",
+      "email" => params["email"] || "email@email#{:rand.uniform(999_999_999_999)}",
+      "colour" => params["colour"] || "#00AA00",
+      "icon" => params["icon"] || "fa-solid fa-user",
+      "password" => params["password"] || "password",
+      "password_confirmation" => params["password"] || "password",
+      "data" => params["data"] || %{}
+    }
+
+    {:ok, _u} =
+      Teiserver.Account.create_user(requested_user)
+  end
+
+  @spec make_users_with_ranks_and_parties(
+          [non_neg_integer()],
+          [non_neg_integer()]
+        ) ::
+          list()
+
+  def make_users_with_ranks_and_parties(list_of_ranks, list_of_parties) do
+    users =
+      Enum.with_index(list_of_ranks)
+      |> Enum.map(fn {_user, index} ->
+        %{
+          "name" => "user" <> to_string(index + 1) <> "_" <> to_string(ExULID.ULID.generate()),
+          "email" => to_string(ExULID.ULID.generate()) <> "@example.com",
+          "permissions" => []
+        }
+      end)
+      |> Enum.map(fn x -> make_user(x) end)
+      # unwrap user creation response
+      |> Enum.map(fn {:ok, reply} -> reply end)
+      |> Enum.map(fn x -> Map.put(x, :userid, x.id) end)
+
+    # set the user rank
+    for {user, rank} <- Enum.zip(users, list_of_ranks) do
+      Teiserver.Account.update_user_stat(user.id, %{
+        :rank => rank
+      })
+    end
+
+    # set the user party_id
+    users =
+      for {user, party_id} <- Enum.zip(users, list_of_parties) do
+        Map.put(user, :party_id, party_id)
+      end
```

Comment:
It seems you are always passing different party ids to this function, and they are all basically indexes.
So you could drop the argument and this loop, and set it when creating the user with the ranks. The entire function can become something like:

```elixir
for {rank, i} <- Enum.with_index(list_of_ranks) do
  {:ok, user} = create_user()
  user = Map.put(user, :party_id, i)
  update_user_stat(...)
  user
end
```

Reply:
True, I guess it could be condensed on the assumption that everyone will be in different parties. I was trying to be explicit-rather-than-clever when trying to write testing code. I also wanted something that was easy to extend if we wanted to test cases with parties ("Oh, let's test this with Alice and Bob the same party, Charlie and Denise in different parties") which could be done with 

`make_users_with_ranks_and_parties([10, 20, 30, 40,], [1, 1, 2, 3])`

Is that valuable at all? 

Happy to throw it out in favor of using `Central.Helpers.GeneralTestLib.make_user()`; just thought I'd offer my reasoning for why I wrote the test helper that way.

Reply:
(resolved in the direction of the requested changes)